src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -50,6 +50,15 @@ public static IFileHandle CreateOutputFile(this IHostEnvironment env, string pat
         }
     }
 
+    [BestFriend]
+    internal interface ICancellableEnvironment
+    {
+        /// <summary>
+        /// Flag which indicate should we stop any code execution in this host.
+        /// </summary>
+        bool IsCancelled { get; }
```

Comment:
`Task` uses the `Canceled` spelling (single `L`):

[https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.iscanceled?view=netframework-4.7.2](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.iscanceled?view=netframework-4.7.2)

Reply:
When in the USA ...

---
In reply to: [264859366](https://github.com/dotnet/machinelearning/pull/2797#discussion_r264859366) [](ancestors = 264859366)